### PR TITLE
Added `CMakeLists.txt` and `Makefile` for clean & fast builds of external projects that use Current.

### PR DESCRIPTION
**Pull request by dimacurrentai**

### CODE REVIEW

**Comment by dimacurrentai** [comment]
Hi @mzhurovich!

This was easier than I thought!

[README](https://github.com/dimacurrentai/Current/blob/cmake/cmake/README.md).

Clean test:

```
$ mkdir clean_current_cmake_test
$ cd clean_current_cmake_test
$ wget https://raw.githubusercontent.com/dimacurrentai/Current/cmake/cmake/run-cmake-test.sh
$ chmod +x run-cmake-test.sh
$ ./run-cmake-test.sh

...

One-line change time, Current gtest, debug: 3s
One-line change time, Current gtest, release: 8s

One-line change time, Google gtest, debug: 3s
One-line change time, Google gtest, release: 2s
```

This test is also run by a Github Action, both on Linux and on macOS.

What this means now is that using Current, with fast incremental builds, and with a non-header-only Googletest, is just a matter of:

* Putting your code under `src/*.cc` (as well as `src/lib_*.cc` and `src/test_*.cc`),
* Grabbing the `Makefile` from `C5T/Current/cmake`, and
* Running `make`, or `make {debug/release/debug_test/release_test/fmt/clean}`.

Thanks,
Dima (from Windows)

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# TODO(dkorolev): Consider intelligently re-running `cmake` from the `Makefile` if the set of source files has changed.
```
Comment (dimacurrentai) [inline cmake/CMakeLists.txt:1]: NOTE(dkorolev), @mzhurovich, this was tested a long time ago in https://github.com/dkorolev/example_use_current_with_cmake, just cleaned up a bit and moved this into `current`.'

**Comment by dimacurrentai** [comment]
@mzhurovich, added support for dynamic libraries too, and tested on Linux, WSL, and macOS!

**Comment by dimacurrentai** [comment]
Probably best to review per commit, but, to get a feel, `cmake/run-cmake-test.sh` is self-contained and self-explanatory!

**Comment by mzhurovich** [comment]
@dimacurrentai  I have one question - why `q/q/q`? =)

**Comment by dimacurrentai** [comment]
It needs to not find current/ in ../ and ../../, and maybe in ../../..

Where did you find it thought? That's my internal local run script!

**Pull request merged**